src/v/cluster/controller_stm.cc

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "cluster/controller_stm.h"
+
+#include "bytes/iostream.h"
+#include "cluster/controller_snapshot.h"
+#include "cluster/logger.h"
+#include "cluster/members_manager.h"
+#include "vlog.h"
+
+namespace cluster {
+
+ss::future<> controller_stm::on_batch_applied() {
+    if (!_feature_table.is_active(features::feature::controller_snapshots)) {
+        co_return;
+    }
+    if (_gate.is_closed()) {
+        co_return;
+    }
+
+    if (
+      last_applied() > _raft->last_snapshot_index()
+      && !_snapshot_debounce_timer.armed()) {
+        _snapshot_debounce_timer.arm(_snapshot_max_age());
+    };
```

Comment:
Should we rearm the timer when snapshot max age changes ?

Reply:
I think it is not worth the complexity - if we already triggered snapshotting, just let it proceed, the changed value will take effect on the next iteration.